cmd/access/staked_access_node_builder.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"time"
+
+	"github.com/onflow/flow-go/cmd"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/module/metrics"
+	"github.com/onflow/flow-go/network/p2p"
+	"github.com/onflow/flow-go/network/topology"
+)
+
+// StakedAccessNodeBuilder builds a staked access node. The staked access node can optionally participate in the
+// unstaked network publishing data for the unstaked access node downstream.
+type StakedAccessNodeBuilder struct {
+	*FlowAccessNodeBuilder
+}
+
+func StakedAccessNode(anb *FlowAccessNodeBuilder) *StakedAccessNodeBuilder {
+	return &StakedAccessNodeBuilder{
+		FlowAccessNodeBuilder: anb,
+	}
+}
+
+// SupportUnstakedNodes returns True if this a staked Access Node which also participates in the unstaked network,
+// False otherwise
+func (builder *StakedAccessNodeBuilder) supportUnstakedNodes() bool {
```

Comment:
I think we should put this function on the parent `FlowAccessNodeBuilder`, and rename to something like `ParticipatesInUnstakedNetwork`.

This is because the `main.go` will need to use this function to determine whether to create a splitter engine and relay engine for a staked access node.

Reply:
done